### ISSUE TITLE
vendor prefixes bug

### ISSUE THREAD

**Issue by lifeiscontent** [CSSLint/csslint#579]
``` css
input[type="search"]::-webkit-search-cancel-button,
input[type="search"]::-webkit-search-decoration { -webkit-appearance: none; }
```

this css line makes this warning trigger.

``` bash
WARNING: The property -moz-appearance is compatible with -webkit-appearance and should be included as well. Include all compatible vendor prefixes to reach a wider range of users. (compatible-vendor-prefixes) Browsers: All
```

css lint should be smarter

**Comment by replete** [comment]
Turn off the vendor prefix warnings. If you're using autoprefixer, or trust yourself to get your inheritance rules right then you don't need it fussing.

**Comment by lifeiscontent** [comment]
@replete this is a webkit specific selector, this has nothing to do with autoprefixer or inheritance, please read the whole code so you don't give invalid suggestions.

**Comment by replete** [comment]
_*puts your toys back in the pram*_

I understand your problem fine. 

Ignoring the vendor-prefix rule is a pragmatic choice if you are trusting in something that handles prefixes for you automatically.

**Comment by lifeiscontent** [comment]
The CSS Linter tool is not supposed to save you from writing bad code, I understand my code.

My colleges however are worse at writing CSS than I am, so I'm forcing them into good convention until they learn. This does NOT fix the problem I am experiencing with process of a company as a whole.

**Comment by replete** [comment]
It seems like you're requiring the passing of all CSSlint tests before allowing code submissions? 

It's a great practice of course, but I find CSSlint a bit too flakey for that! You can't even really ignore chunks of CSS causing false-positives.

This is a frustration for me too. I have resorted to using CSSlint as a code quality reminder rather than an enforcement of good standards. 

Anyway, I hope someone can look into this for you.

**Comment by stephenbunch** [comment]
@replete So I think the problem is that the element being styled is specific to webkit. It doesn't make sense to warn about supporting other browsers when the selector in question is browser specific. In this example, `-moz-appearance` wouldn't have any effect because the selector only applies to webkit.

**Comment by stubbornella** [comment]
This seems like a reasonable request for a PR.

**Comment by stubbornella** [comment]
Roughly, apply vendor prefix checks only when in a generic (non vendor prefixed) selector.

**Comment by lifeiscontent** [comment]
@stubbornella exactly! :)

**Comment by stubbornella** [comment]
Aaron, do you want to submit a PR?

**Comment by lifeiscontent** [comment]
@stubbornella Sure, I'll try to take a stab at it sometime tomorrow.

**Comment by rhutchison** [comment]
Any fix yet?
